validphys2/src/validphys/commondataparser.py

```diff
@@ -41,20 +83,585 @@
+    """
+    data_file = metadata.path_data_central
+    datayaml = yaml.safe_load(data_file.read_text(encoding="utf-8"))
+    data_df = pd.DataFrame(
+        datayaml["data_central"], index=range(1, metadata.ndata + 1), columns=["data"]
+    )
+    data_df.index.name = _INDEX_NAME
+    return data_df
+
+
+def _parse_uncertainties(metadata):
+    """Given the metadata defining the commondata,
+    returns a dataframe with all appropiate uncertainties
+
+    Parameters
+    ----------
+    metadata: ObservableMetaData
+        instance of ObservableMetaData defining the uncertainties to be loaded
+
+    Returns
+    -------
+    pd.DataFrame
+        a dataframe containing the uncertainties
+    """
+    all_df = []
+    for ufile in metadata.paths_uncertainties:
+        uncyaml = yaml.safe_load(ufile.read_text())
+
+        mindex = pd.MultiIndex.from_tuples(
+            [(k, v["treatment"], v["type"]) for k, v in uncyaml["definitions"].items()],
+            names=["name", "treatment", "type"],
+        )
+        # I'm guessing there will be a better way of doing this than calling  dataframe twice for the same thing?
+        final_df = pd.DataFrame(
+            pd.DataFrame(uncyaml["bins"]).values,
+            columns=mindex,
+            index=range(1, metadata.ndata + 1),
+        )
+        final_df.index.name = _INDEX_NAME
+        all_df.append(final_df)
+    return pd.concat(all_df, axis=1)
+
+
+def _parse_kinematics(metadata):
+    """Given the metadata defining the commondata,
+    returns a dataframe with the kinematic information
+
+    Parameters
+    ----------
+    metadata: ObservableMetaData
+        instance of ObservableMetaData defining the kinematics to be loaded
+
+    Returns
+    -------
+    pd.DataFrame
+        a dataframe containing the kinematics
+    """
+    kinematics_file = metadata.path_kinematics
+    kinyaml = yaml.safe_load(kinematics_file.read_text())
+
+    kin_dict = {}
+    for i, dbin in enumerate(kinyaml["bins"]):
+        bin_index = i + 1
+        # TODO: for now we are dropping min/max information since it didn't exist in the past
+        # unless the point doesn't have a mid value, in that case we need to generate it!
+        for d in dbin.values():
+            if d["mid"] is None:
+                d["mid"] = 0.5 * (d["max"] + d["min"])
+            d["min"] = None
+            d["max"] = None
+        kin_dict[bin_index] = pd.DataFrame(dbin).stack()
+
+    return pd.concat(kin_dict, axis=1, names=[_INDEX_NAME]).swaplevel(0, 1).T
+
+
+def parse_new_metadata(metadata_file, observable_name, variants=[]):
+    """Given a metadata file in the new format and the specific observable to be read
+    load and parse the metadata and select the observable.
+    If any variants are selected, apply them.
+    """
+    # Note: we are re-loading many times the same yaml file, possibly a good target for lru_cache
+    set_metadata = parse_yaml_inp(metadata_file, SetMetaData)
```

Comment:
```suggestion
    metadata_file = Path(metadata_file)
    set_metadata = parse_yaml_inp(metadata_file, SetMetaData)
```

Otherwise if `metadata_file` is a standard str pointing at the path of metadata.yaml, the parser would crush since strings have no `.parent` attribute